addons/l10n_sa_edi/views/report_invoice.xml

```diff
@@ -0,0 +1,139 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+
+        <template id="arabic_english_invoice" inherit_id="l10n_sa_invoice.arabic_english_invoice">
+
+            <!--    Add Currency Exchange rate if different currency than SAR    -->
+            <xpath expr="//div[hasclass('clearfix')]" position="after">
+                <div t-if="o.company_id.country_id.code == 'SA' and o.currency_id != o.company_id.currency_id"
+                     id="sar_amounts" class="row clearfix ml-auto my-3 text-nowrap table">
+                    <t t-set="sar_rate"
+                       t-value="o.env['res.currency']._get_conversion_rate(o.currency_id, o.company_id.currency_id, o.company_id, o.l10n_sa_confirmation_datetime)"/>
```

Comment:
Did not we add this recently in 16.0 and could not we use the same code from there?  (to avoid the _get_conversion_rate thing)